mindmeld/auto_annotator.py

```diff
@@ -835,5 +901,320 @@ def valid_entity_check(self, entity):
+            value_matched_candidates = []
+            for candidate in candidates:
+                if entity["dim"] != candidate["entity_type"]:
+                    continue
+                if entity["value"] == candidate["value"]:
+                    value_matched_candidates.append(candidate)
+                if value_matched_candidates:
+                    continue
+                if (
+                    self.translator.translate(
+                        entity["body"], target_language=self.language
+                    )
+                    == candidate["body"]
+                ):
+                    final_candidates.append(candidate)
+                    break
+            if value_matched_candidates:
+                final_candidates.append(
+                    max(value_matched_candidates, key=lambda x: len(x["body"]))
+                )
+        if entity_types:
+            final_candidates = [
+                e for e in final_candidates if e["entity_type"] in entity_types
+            ]
+        return [
+            duckling_item_to_entity_dict(candidate) for candidate in final_candidates
+        ]
+
+    @property
+    def supported_entity_types(self):  # pylint: disable=W0236
+        """
+        Returns:
+            supported_entity_types (list): List of supported entity types.
+        """
+        supported_entity_types = set(
+            self.en_annotator.supported_entity_types
+        ).intersection(DUCKLING_TO_SYS_ENTITY_MAPPINGS[self.language])
+        return list(supported_entity_types)
+
+
+class MultiLingualAnnotator(Annotator):
+    """Custom Annotator class used to generate annotations."""
+
+    def __init__(self, *args, **kwargs):
+        """Initializes a TranslationDucklingAnnotator.
+
+        Args:
+            app_path (str): The location of the MindMeld app.
+            annotation_rules (list): List of Annotation rules.
+            en_annotator (SpacyAnnotator): A Spacy Annotator with language set to English ("en").
+            translator (str): A translator to use such as 'GoogleTranslator' or 'NoOpTranslator'.
+            language (str, optional): Language as specified using a 639-1/2 code.
+            locale (str, optional): The locale representing the ISO 639-1 language code and \
+                ISO3166 alpha 2 country code separated by an underscore character.
+            overwrite (bool): Whether to overwrite existing annotations with conflicting spans.
+            unannotate_supported_entities_only (bool): Only allow removal of supported entities.
+            unannotation_rules (list): List of Annotation rules.
+        """
+        super().__init__(*args, **kwargs)
+        self.translator = kwargs.get("translator", "NoOpTranslator")
+        self.en_annotator = SpacyAnnotator(
+            app_path=self.app_path,
+            language=ENGLISH_LANGUAGE_CODE,
+            locale=ENGLISH_US_LOCALE,
+        )
+        if self.language != ENGLISH_LANGUAGE_CODE:
```

Comment:
Does it makes sense for anyone to use `language=en` here? The MLA seems to be use for only non-english usecases, maybe we should enforce that?

Reply:
The MLA is currently designed to work for all languages, both English and Non-English. This makes it so that developers don’t confused on when to use a SpacyAnnotator versus MLA. Under the hood, MLA is returning the output of a SpacyAnnotator if the language is English.